crates/vqueues/src/scheduler/drr.rs

```diff
@@ -333,14 +370,33 @@ where
                 let Some(qstate) = self.q.get_mut(&qid) else {
                     return Ok(());
                 };
+                let config = vqueues.config_pool().find(&qid.parent);
                 let meta = vqueues.get_vqueue(&qid).unwrap();
 
                 if qstate.remove_from_unconfirmed_assignments(item_hash) {
                     // drop the already acquired permit
                     let _ = self.unconfirmed_capacity_permits.split(1);
-                } else {
-                    qstate.notify_removed(item_hash);
+                } else if qstate.notify_removed(item_hash) {
+                    match qstate.check_eligibility(now, meta, config) {
+                        Eligibility::Eligible if !self.eligible.contains(&qid) => {
+                            // Make eligible immediately.
+                            qstate.deficit.set_last_round(self.global_sched_round);
+                            self.eligible.push_back(qid);
+                            self.waker.wake_by_ref();
+                        }
```

Comment:
What's the case where removing the head element will make a vqueue eligible which wasn't eligible before? Or is more to be defensive here in case this will become a case in the future?